File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        final long low = Long.MIN_VALUE;
        long first = low, second = low, third = low;
        final int n = nums.length;
        for(int i = 0; i < n; i++) {
            long curr = (long) nums[i];
            if(curr > first) {
                third = second;
                second = first;
                first = curr;
            } else if(curr < first && curr > second) {
                third = second;
                second = curr;
            } else if(curr < second && curr > third) {
                third = curr;
            }
        }
        
        return third == low ? (int) first : (int) third;
        
    }
}

File: 0414-third-maximum-number/0414-third-maximum-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var thirdMax = function(nums) {
    
    // let arr = [];
    // let set = new Set();
    // for(let num of nums) {
    //     if(set.has(num)) {
    //         continue;
    //     }
    //     arr.push(num);
    // }
    
    let first = -Infinity;
    let second = -Infinity;
    let third = -Infinity;
    
    for(let num of nums) {
        if(num > first) {
            third = second;
            second = first;
            first = num;
        } else if(num < first && num > second) {
            third = second;
            second = num;
        } else if(num < second && num > third) {
            third = num;
        }
    }
    
    return third == -Infinity ? first : third;
    
};